packages/js/src/executors/prune-lockfile/prune-lockfile.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  detectPackageManager,
+  type ExecutorContext,
+  logger,
+  parseTargetString,
+  type ProjectGraph,
+  readJsonFile,
+  workspaceRoot,
+} from '@nx/devkit';
+import { existsSync, lstatSync, writeFileSync } from 'fs';
+import { dirname, join } from 'path';
+import { interpolate } from 'nx/src/tasks-runner/utils';
+import { type PackageJson } from 'nx/src/utils/package-json';
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import {
+  getLockFileName,
+  createLockFile,
+} from 'nx/src/plugins/js/lock-file/lock-file';
+import { type PruneLockfileOptions } from './schema';
+
+export default async function pruneLockfileExecutor(
+  schema: PruneLockfileOptions,
+  context: ExecutorContext
+) {
+  logger.log('Pruning lockfile...');
+  const outputDirectory = getOutputDir(schema, context);
+  const packageJson = getPackageJson(schema, context);
+  const { lockfileName, lockFile } = createPrunedLockfile(
+    packageJson,
+    context.projectGraph
+  );
+  writeFileSync(join(outputDirectory, lockfileName), lockFile);
+  writeFileSync(
+    join(outputDirectory, 'package.json'),
+    JSON.stringify(packageJson, null, 2)
+  );
+  logger.log('Lockfile pruned.');
+  logger.log('Success!');
```

Comment:
Minor: I think prune take that long, so reducing the log to just one at the end, like `Lockfile pruned: ${output}` makes more sense.